### ISSUE TITLE
[Bug]: Stage rewards don't properly refresh JEI

### ISSUE THREAD

**Issue by vizthex123** [FTBTeam/FTB-Mods-Issues#975]
### Mod

FTB Quests

### Mod version

ftb-quests-forge-1902.5.6-build.304.jar

### Forge / Fabric version

Forge 43.3.2

### What issue are you having?

When a quest reward is set to give a GameStage, it will grant the stage but not update the player's data so they can view the unlocked items in JEI. Reloading the world fixes it.

Game log:
[latest.log](https://github.com/FTBTeam/FTB-Mods-Issues/files/12874972/latest.log)

### Steps to reproduce

1. Claim a stage reward
2. Check JEI for any item staged behind it

**Comment by desht** [comment]
I suspect the issue here is that you're using the Item Stages mod but not in conjunction with Game Stages.  From your log file:

```
[15:07:19] [modloading-worker-0/INFO]: Chose [KubeJS Stages] as the active game stages implementation
```

FTB Quests (or more precisely FTB XMod Compat) will use KJS in preference to Game Stages as its staging implementation if both mods are installed. I haven't looked at the Item Stages code in detail, but I suspect KJS's implementation doesn't trigger it in the same way.

I guess what's really needed here is a way (in mod config) to specify a particular stages implementation to be used when more than one is available...

**Comment by vizthex123** [comment]
> I suspect the issue here is that you're using the Item Stages mod but not in conjunction with Game Stages. From your log file:
> 
> ```
> [15:07:19] [modloading-worker-0/INFO]: Chose [KubeJS Stages] as the active game stages implementation
> ```
> 
> FTB Quests (or more precisely FTB XMod Compat) will use KJS in preference to Game Stages as its staging implementation if both mods are installed. I haven't looked at the Item Stages code in detail, but I suspect KJS's implementation doesn't trigger it in the same way.
> 
> I guess what's really needed here is a way (in mod config) to specify a particular stages implementation to be used when more than one is available...

nope, both are installed (have been since I made the pack). Odd that it isn't being picked up though...

![image](https://github.com/FTBTeam/FTB-Mods-Issues/assets/31721920/58c3134b-9fa7-4a92-8ff5-e506bd34428c)

but yeah, having a config option would be nice. Didn't even know KubeJS could do it since their docs are so archaic (barely got custom items working and that's all i use it for).

**Comment by desht** [comment]
I think you missed my comment:

> FTB Quests (or more precisely FTB XMod Compat) will use KJS in preference to Game Stages as its staging implementation if both mods are installed

Right now, even when both KJS and Game Stages are installed, it's always KJS stages which get used.  That's something I am working on for the next FTB XMod Compat release - a config setting to explicitly choose to use Game Stages in this sort of situation.
